### PR TITLE
refactor(react-instantsearch): migrate from react-instantsearch to react-instantsearch-dom and react-instantsearch-core

### PR DESCRIPTION
**Pull request by Haroenv**
## What kind of change does this PR introduce?

Updates the packages of react-instantsearch to react-instantsearch-dom and -core. Also updates the types dependencies to avoid relying on transitive dependencies.

## What is the current behavior?

Things are imported mixed from react-instantsearch, /dom, -dom and -core

## What is the new behavior?

Things are only imported from -core and -dom

## What steps did you take to test this? This is required before we can merge, make sure to test the flow you've updated.

1. updated the imports
2. updated the dependency

To be done in review is checking whether typescript etc. is properly updated as well, as that didn't seem to work locally for me. Possibly there was some flow code left that needs to be updated too (although i didn't see any)

## Checklist

- [x] Documentation
- [ ] Testing 
- [x] Ready to be merged
      
- [x] Added myself to contributors table

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/5953/commits/35c80cfa88396a85ea3a370065c1ef0d9a74eb6e failed.

File: yarn.lock [hunk 1]
```diff
@@ -26061,7 +26208,7 @@ qrcode.react@^0.8.0:
     prop-types "^15.6.0"
     qr.js "0.0.0"
 
-qs@6.7.0, qs@^6.5.0, qs@^6.5.1, qs@^6.6.0:
+qs@6.7.0, qs@^6.5.0, qs@^6.6.0, qs@^6.5.1::
```
Comment (Haroenv) [inline yarn.lock:26211]:
```suggestion
qs@6.7.0, qs@^6.5.0, qs@^6.6.0, qs@^6.5.1:
```

**Comment by Haroenv** [comment]
```


../../node_modules/@types/react-instantsearch-core/index.d.ts(12,34): error TS7016: Could not find a declaration file for module 'algoliasearch-helper'. '/drone/src/node_modules/algoliasearch-helper/index.js' implicitly has an 'any' type.
--
40 | Try `npm install @types/algoliasearch-helper` if it exists or add a new declaration (.d.ts) file containing `declare module 'algoliasearch-helper';`
41

```

This error on drone is odd, as @types/algoliasearch-helper *is* installed

**Comment by lbogdan** [comment]
It errors out  on all 3 CIs, with the same error. Does it work if you build it (`yarn build`) locally?

**Comment by Haroenv** [comment]
It fails locally too, but I'm not sure why that even checks in the -api folder, it doesn't import React InstantSearch 🤔 is there some global file that could be the cause @lbogdan ?

**Comment by lbogdan** [comment]
No idea, maybe @CompuIves can help?

File: packages/app/package.json [hunk 2]
```diff
@@ -285,7 +286,8 @@
     "@types/react-dom": "^16.8.3",
     "@types/react-helmet": "^5.0.11",
     "@types/react-icons": "2.2.7",
-    "@types/react-instantsearch": "^5.2.3",
+    "@types/react-instantsearch-dom": "^5.2.3",
+    "@types/react-instantsearch-core": "^5.2.3",
```
Comment (Haroenv) [inline packages/app/package.json:290]: maybe it works if the changes in package.json are undone (it should also work in theory thanks to the transitive dependencies used

**Comment by alexnm** [comment]
Since this was created, we shifted around some of the components in the create sandbox modal, do you mind redoing this PR if the change is still relevant? the react search functionality remained largely the same

**Pull request closed without merging**